ow_regolith/include/DigStateMachine.h

```diff
@@ -0,0 +1,176 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+// Defines a finite state machine and associated states that estimate the phase
+// of a digging operation performed with the lander scoop.
+
+#ifndef DIG_FSM_H
+#define DIG_FSM_H
+
+#include <ros/ros.h>
+#include <tf/tf.h>
+
+#include <string>
+#include <memory>
+
+namespace ow_regolith {
+
+class DigStateMachine;
+
+class DigState
+{
+public:
+  DigState(const std::string &name, DigStateMachine *const c)
+    : m_name(name), m_context(c) { };
+  virtual ~DigState() = default;
+
+  DigState() = delete;
+  DigState(const DigState&) = delete;
+  DigState& operator=(const DigState&) = delete;
+
+  inline std::string getName() const {return m_name;}
```

Comment:
Minor: functions defined in the class should be implicitly 'inline' without the keyword -- this has always been my understanding.